File: app/src/main/java/com/example/sachinsandbhor/retrofitsample/presentation/movieList/MovieListContract.java
package com.example.sachinsandbhor.retrofitsample.presentation.movieList;

import com.example.sachinsandbhor.retrofitsample.MVP;
import com.example.sachinsandbhor.retrofitsample.model.MovieResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by Sachin.Sandbhor on 06-02-2018.
 */

public interface MovieListContract {

    interface MovieListView extends MVP.View{

        void setMovieList(MovieResponse movieResponse);

        void startLoadingPost();

        void stopLoadingPost();

        void navigateToMovieDetail(int id);

        void onError(String errorMessage);

    }

    interface MovieListPresenter extends MVP.Presenter{

        void getMovieList();

        void navigateToMovieDetail(int id);

        void detach();

        void onError(String errorMessage);

        void setView(MovieListContract.MovieListView movieListView);

    }

    interface MovieListModel extends MVP.Model{

        void getMovieList();

        interface MovieListPresenterCallBack {

            void onMovieListReceived(MovieResponse movieResponse);

            void onError(String errorMessage);
        }

    }

    interface apiServices {

        @GET("movie/top_rated")
        Observable<MovieResponse> getRatedMovies(@Query("api_key") String apiKey);
    }

}

File: retrofitkotlin/src/main/java/com/sachinsandbhor/retrofitkotlin/presentation/movielist/MovieListContract.kt
package com.sachinsandbhor.retrofitkotlin.presentation.movielist

import com.sachinsandbhor.retrofitkotlin.MVPInterface
import com.sachinsandbhor.retrofitkotlin.domain.MovieListResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by Sachin.Sandbhor on 07-03-2018.
 */
interface MovieListContract {

    interface MovieListView:MVPInterface.MVPView {
        fun startLoading()
        fun stopLoading()
        fun setMovie(movieListResponse :MovieListResponse)
        fun onError(message:String?)
        fun navigateUser(id:Int)
    }

    interface MovieListPresenter:MVPInterface.MVPPresenter{
        fun getMovieList(pageNo:Int)
        fun setView(view: MovieListContract.MovieListView)
        fun navigateUser(id:Int)
    }

    interface MovieListModel:MVPInterface.MVPModel{
        fun getMovieList(pageNo:Int)
        interface MovieListCallback{
            fun movieListReceived(movieResponse: MovieListResponse)
            fun onError(message: String?)
        }
    }

    interface apiService{
        @GET("movie/top_rated")
        fun getRatedMovies(@Query("api_key") apiKey: String, @Query ("page") page:Int): Observable<MovieListResponse>
    }
}